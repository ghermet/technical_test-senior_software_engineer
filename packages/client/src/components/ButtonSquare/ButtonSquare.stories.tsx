import React from 'react'
import ButtonSquare from './ButtonSquare'
import { ChevronRightBold } from '@pretto/picto'
import '../../styles/globals.generated.css'

export default {
  title: 'Components/ButtonSquare',
  component: ButtonSquare,
}

export const Default = () => (
  <ButtonSquare>
    <ChevronRightBold />
  </ButtonSquare>
)

export const Disabled = () => (
  <ButtonSquare disabled>
    <ChevronRightBold />
  </ButtonSquare>
)
