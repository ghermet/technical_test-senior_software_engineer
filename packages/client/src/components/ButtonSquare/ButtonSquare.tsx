import React from 'react'
import { ctwMerge } from '../../utility/ctw-merge'

export type ButtonSquareProps = React.DetailedHTMLProps<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>

const ButtonSquare = ({ children, className, ...props }: ButtonSquareProps) => (
  <button
    {...props}
    className={ctwMerge(
      'items-center appearance-none bg-accent1 rounded text-white cursor-pointer outline-offset-2 outline-dashed inline-flex h-8 justify-center align-middle w-8 p-0 border-none transition-colors duration-150 ease-in-out',
      'hover:bg-accent1Hover',
      'focus:outline-accent1Hover focus:outline-2',
      'focus-visible:bg-accent1Hover focus-visible:outline-accent1Hover focus-visible:outline-2',
      'disabled:bg-accent1Disabled disabled:cursor-not-allowed',
      className
    )}
  >
    {children}
  </button>
)

export default ButtonSquare
