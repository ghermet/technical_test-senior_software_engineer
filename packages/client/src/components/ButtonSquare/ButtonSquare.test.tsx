import React from 'react'
import { render } from '@testing-library/react'
import ButtonSquare from './ButtonSquare'

describe('ButtonSquare', () => {
  it('should render a button element with the correct default classnames and props', () => {
    const { container } = render(<ButtonSquare className="some-class">Submit</ButtonSquare>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a disabled button element', () => {
    const { getByRole } = render(<ButtonSquare disabled>Submit</ButtonSquare>)
    const button = getByRole('button') as HTMLButtonElement
    expect(button.disabled).toBe(true)
  })
  
  it('should call `onClick` when clicking button', () => {
    const onClickMock = jest.fn()
    const { getByRole } = render(<ButtonSquare onClick={onClickMock}>Submit</ButtonSquare>)
    getByRole('button').click()
    expect(onClickMock).toHaveBeenCalled()
  })
})
