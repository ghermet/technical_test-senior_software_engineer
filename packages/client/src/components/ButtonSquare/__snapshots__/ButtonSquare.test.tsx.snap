// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSquare should render a button element with the correct default classnames and props 1`] = `
<button
  class="items-center appearance-none bg-accent1 rounded text-white cursor-pointer outline-offset-2 outline-dashed inline-flex h-8 justify-center align-middle w-8 p-0 border-none transition-colors duration-150 ease-in-out hover:bg-accent1Hover focus:outline-accent1Hover focus:outline-2 focus-visible:bg-accent1Hover focus-visible:outline-accent1Hover focus-visible:outline-2 disabled:bg-accent1Disabled disabled:cursor-not-allowed some-class"
>
  Submit
</button>
`;
